### PR TITLE
WIP: added web push notifications for download completion

### PR DESCRIPTION
**Pull request by abhinadduri**

### CODE REVIEW

File: public/serviceWorker.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+self.addEventListener('push', function(event) {
+  event.waitUntil(
+    self.registration.showNotification('Firefox Send', {
+      body: event.data.text() + ' has been downloaded.',
```
Comment (pdehaan) [inline public/serviceWorker.js]: Does this need to be localized?

File: frontend/src/upload.js [hunk 2]
```diff
@@ -26,6 +26,34 @@ if (storage.has('referrer')) {
...
+      .then(function(subscription) {
+        if (subscription) {
+          return subscription;
+        }
+        return registration.pushManager.subscribe({ userVisibleOnly: true });
+      });
+    }).then(function(subscription) {
+      const rawKey = subscription.getKey ? subscription.getKey('p256dh') : '';
+      const key = rawKey ?
+        btoa(String.fromCharCode.apply(null, new Uint8Array(rawKey))) : '';
```
Comment (pdehaan) [inline frontend/src/upload.js]: Only duplicated twice, but not sure if you want to wrap this in a helper function.

File: server/server.js [hunk 3]
```diff
@@ -177,6 +178,17 @@ app.get('/assets/download/:id', (req, res) => {
...
+            webPush.sendNotification({
+              endpoint: meta.endpoint,
+              keys: {
+                p256dh: meta.key,
+                auth: meta.authSecret
+              }
+            },
+            meta.filename)
+            .catch(err => {
+              console.log(err);
```
Comment (pdehaan) [inline server/server.js]: Do we need to log this error somewhere? Or is it a non-important error?

Comment (abhinadduri) [inline server/server.js]: It's a non-important error. I actually meant to use `.catch(err=>0)`, but I put it there for debugging purposes for now.

File: public/serviceWorker.js (hunk 1 above)
Comment (abhinadduri) [inline public/serviceWorker.js]: Yeah, nice catch. We'll have to wrap this in a promise reaching out to l20n.

**Pull request closed without merging**